src/aeso_ast_infer_types.erl

```diff
@@ -626,7 +630,9 @@ infer_contract_top(Env, Kind, Defs0, _Options) ->
 %% a list of definitions.
 -spec infer_contract(env(), main_contract | contract | namespace, [aeso_syntax:decl()]) -> {env(), [aeso_syntax:decl()]}.
 infer_contract(Env0, What, Defs0) ->
+    create_type_errors(),
     Defs = process_blocks(Defs0),
+    destroy_and_report_type_errors(Env0),
```

Comment:
@UlfNorell could you verify? I could not leave these type errors undestroyed here because it was causing the whole compiler to crash, had to create separate ones for this particular case.

Reply:
lgtm